### ISSUE TITLE
Enable to disable editing on per cell basis   (via TableInlineCellEditing and DataTypeProvider)

### ISSUE THREAD

**Issue by Carduelis** [DevExpress/devextreme-reactive#3021]
> and how will it work with custom DataTypeProvider? Beacuse I've tried to implement your sample into already working DataTypeProvider and it seems not to catch the editingEnabled set in this TableEditRow.Cell, but if I use default type provider it works perfectly

> Currently, this case is not available. We will discuss how to implement it in the future.

Hello! My case is to use ability to disable ability to edit cells on per-cell basis. We use `TableInlineCellEditing` component.

When I put `<DataTypeProvider editorComponent={CustomEditor} />` it has always `disabled: false` property nethertheless I put `editingEnabled={false}`, but when I remove the DataTypeProvider, and the default editor is shown -- it renders as disabled.

My goal is to be able to set disability on per cell base editing.

```
const Cell = ({ row, column, editingEnabled, ...restProps }) => {
  return (
    <TableInlineCellEditing.Cell
      row={row}
      column={column}
      editingEnabled={false}
      {...restProps}
    />
  )
}
```

_Originally posted by @MaximKudriavtsev in https://github.com/DevExpress/devextreme-reactive/issues/953#issuecomment-478026511_

**Comment by LazyLahtak** [comment]
Hi @Carduelis,

The [DataTypeProvider.ValueEditor](https://devexpress.github.io/devextreme-reactive/react/grid/docs/reference/data-type-provider/#datatypeprovidervalueeditorprops) component's `disabled` property and the [InlineCellEditing.Cell](https://devexpress.github.io/devextreme-reactive/react/grid/docs/reference/table-inline-cell-editing/#tableinlinecelleditingcellprops) component's `editingEnabled` property don't affect each other. These values are defined by the `columnExtensions` [property](https://devexpress.github.io/devextreme-reactive/react/grid/docs/reference/editing-state/#editingstatecolumnextension) of the `EditingState` plugin.
Could you please clarify your scenario? What is the purpose of using a custom `InlineCellEditing.Cell` component with `DataTypeProvider`?

**Comment by Carduelis** [comment]
> What is the purpose of using a custom InlineCellEditing.Cell component with DataTypeProvider?

I checked how the `DataTypeProvider` formatter/editorComponent recieves `disabled=true/false` -- the reason is flag `editingEnabled`. 

I found a way how to pass on per-cell basis. This is why I put this flag to the custom `InlineCellEditing.Cell` component. 
Unfortunately, it works only if I remove `DataTypeProvider` with my custom formatters/editors. 

> Could you please clarify your scenario?
It's easy: our goal is disabling editing on per cell basis. In this case the `formatterComponent` should not be switched to the `editorComponent`. Or another approach to make it. (default behavior with InlineEditing is scary: if the cell is disabled (example - the column is disabled from editing). and user knows that the cell is readonly **only when** he clicks on the cell. Which is not the best UX: user should know that the cell is not for editing even when it's a `formatterComponent`.

Current behavior of Devexpress is: if we go to edit-mode of the row, we can mark some columns as disabled. In this case the row renders all of the inputs, but some of them are disabled. 
There is no way to set it on per-cell basis.
